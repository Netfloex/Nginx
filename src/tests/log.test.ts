import { replaceCurrentDir } from "./test-utils";
import { performance } from "perf_hooks";

import { logger } from "@lib/logger";

describe("Logger", () => {
	logger.overWriteLogFunction = jest.fn().mockName("logger");
	Object.defineProperty(performance, "now", {
		value: jest.fn(() => 0).mockName("performance.now")
	});
	const mockedLog = jest.mocked(logger.overWriteLogFunction);
	test("All log Messages", () => {
		Object.values(logger).forEach((log) => log({} as never));
		expect(mockedLog).toBeCalledTimes(Object.values(logger).length);

		expect(
			mockedLog.mock.calls.flat().map((msg) => replaceCurrentDir(msg))
		).toMatchSnapshot();
	});
});
