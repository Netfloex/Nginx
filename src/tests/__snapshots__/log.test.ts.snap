// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger All log Messages 1`] = `
[
  "[NCM] [INFO]               Started",
  "[NCM] [DONE]               Done in 0.000s",
  "[NCM] [ERROR]              Exited in 0.000s",
  "[NCM] [ERROR]              An unexpected error occurred:",
  "[NCM] [INFO]  [NGINX]      Removing old generated configs... /current/nginx/conf.d",
  "[NCM] [ERROR] [NGINX]      Nginx Config Path not found: /current/nginx/conf.d",
  "[NCM] [INFO]  [NGINX]      You can set NGINX_CONFIG_PATH env variable to customize this location.",
  "[NCM] [WARN]  [NGINX]      Could not find nginx.conf: undefined",
  "[NCM] [ERROR] [CERTBOT]    You must set the CERTBOT_EMAIL environment variable, certbot can not run without it.",
  "[NCM] [ERROR] [CERTBOT]    Certbot binary not found",
  "[NCM] [INFO]  [CERTBOT]    All domains have their valid certificates, skipping certbot.",
  "[NCM] [INFO]  [CERTBOT]    Certbot is disabled.",
  "[NCM] [INFO]  [CERTBOT]    Skipped requesting certificates",
  "[NCM] [INFO]  [CERTBOT]    Requesting certificates for -1 domains...",
  "[NCM]         [CERTBOT]    [0/-1] undefined: undefined",
  "[NCM] [ERROR] [CERTBOT]    Certbot ran into an error:
 ",
  "[NCM] [ERROR] [CERTBOT]    Tried renewing a production certificate (undefined) while in staging mode, this is not a good idea.",
  "[NCM] [INFO]  [CERTBOT]    The server undefined has missing certificate files, requesting...",
  "[NCM] [ERROR] [CERTBOT]    The certificate files for undefined could not be renewed because CERTBOT_EMAIL is not set. This domain is now disabled.",
  "[NCM] [INFO]  [CERTBOT]    The certificate for undefined is valid for -1 days",
  "[NCM] [INFO]  [CERTBOT]    The certificate for undefined, expired 1 days ago",
  "[NCM] [INFO]  [CERTBOT]    The certificate for undefined is using the Staging environment, renewing for production.",
  "[NCM] [ERROR] [CERTBOT]    Parsing of the certificate undefined failed: undefined",
  "[NCM] [INFO]  [DHPARAMS]   Creating a 2048 bit Diffie-Hellman parameter... This could take a while.",
  "[NCM] [DONE]  [DHPARAMS]   Diffie-Hellman parameters are created successfully, took NaN seconds. /etc/letsencrypt/dhparams/dhparam.pem",
  "[NCM] [ERROR] [DHPARAMS]   Could not find the letsencrypt directory: /etc/letsencrypt",
  "[NCM] [ERROR] [DHPARAMS]   So the Diffie-Hellman parameter could not be created at /etc/letsencrypt/dhparams/dhparam.pem",
  "[NCM] [DONE]  [NGINX]      undefined",
  "[NCM] [ERROR] [CONFIG]     There is no config file in:  undefined",
  "[NCM] [ERROR] [CONFIG]     The config folder was not found: undefined",
  "[NCM] [WARN]  [CONFIG]     Multiple config files were found, using undefined",
  "[NCM] [ERROR] [CONFIG]     There was an issue with your config, the error is listed below.",
  "[NCM] [ERROR] [CONFIG]     There was an error parsing the config, please check undefined",
  "[NCM] [ERROR] [CONFIG]     Yaml Error: undefined:
undefined",
  "[NCM] [ERROR] [CONFIG]     JSON Error: undefined",
  "[NCM] [ERROR] [CONFIG]     JS Error: undefined",
  "[NCM] [ERROR] [CONFIG]     The JS file returned undefined, expected ",
  "[NCM] [ERROR] [CONFIG]     The config is empty. undefined",
  "[NCM] [ERROR] [CONFIG]     Path: config
Error: undefined",
  "[NCM] [INFO]  [CONFIG]     Config is valid undefined",
  "[NCM] [WARN]  [CONFIG]     Config contains undefined, while process.env.undefined was not defined.",
  "[NCM] [INFO]  [CONFIG]     Config is a promise, waiting until it resolves.",
  "[NCM] [WARN]  [CONFIG]     Could not resolve undefined, normally this would exit.",
  "[NCM] [WARN]  [CONFIG]     The path undefined could not be found, but since you are using the standalone version you might not need it in this container.",
  "[NCM] [INFO]  [CSS]        Downloading CSS file... undefined",
  "[NCM] [DONE]  [CSS]        CSS file is cached, skipping: undefined",
  "[NCM] [DONE]  [CSS]        Downloaded CSS file undefined",
  "[NCM] [ERROR] [CSS]        There was an error downloading/compressing the url: undefined
undefined",
  "[NCM] [ERROR] [CSS]        Could not save the CSS file undefined
undefined",
  "[NCM] [INFO]  [JS]         Downloading JS file... undefined",
  "[NCM] [DONE]  [JS]         JS file is already downloaded, skipping: undefined",
  "[NCM] [DONE]  [JS]         Downloaded JS file undefined",
  "[NCM] [ERROR] [JS]         undefined",
  "[NCM] [INFO]  [CLOUDFLARE] Updating Cloudflare ip list...",
  "[NCM] [INFO]  [CLOUDFLARE] Using cache: current ip list has been updated today",
  "[NCM] [INFO]  [CLOUDFLARE] Cloudflare ip list expired, Cache duration: 7 days",
  "[NCM] [DONE]  [CLOUDFLARE] Updated Cloudflare ip list Request took NaNs",
  "[NCM] [INFO]  [CLOUDFLARE] Cloudflare ip list refreshed, no changes with cache Request took NaNs",
  "[NCM] [DONE]  [CLOUDFLARE] Cloudflare ip list has been generated. Added undefined ip addresses.",
  "[NCM] [ERROR] [CLOUDFLARE] undefined",
  "[NCM] [WARN]  [ENV]        Could not parse undefined to a number, defaulting to undefined",
  {},
]
`;
